### PR TITLE
remove outdated `FIXME`s in bootstrap internals

### PR DESCRIPTION
**Pull request by onur-ozkan**
self-explanatory

### CODE REVIEW

File: src/bootstrap/bootstrap.py [hunk 1]
```diff
@@ -916,10 +916,6 @@ def build_bootstrap_cmd(self, env):
         env.setdefault("RUSTFLAGS", "")
         # we need to explicitly add +xgot here so that we can successfully bootstrap
         # a usable stage1 compiler
-        # FIXME: remove this if condition on the next bootstrap bump
-        # cfg(bootstrap)
-        if self.build_triple().startswith('mips'):
-            env["RUSTFLAGS"] += " -Ctarget-feature=+xgot"
```
Comment (ozkanonur) [inline src/bootstrap/bootstrap.py:922]: https://github.com/rust-lang/rust/pull/113637

File: src/bootstrap/bootstrap.py [hunk 2]
```diff
@@ -916,10 +916,8 @@ def build_bootstrap_cmd(self, env):
...
+        env["RUSTFLAGS"] += " -Ctarget-feature=+xgot"
+
```
Comment (ozkanonur) [inline src/bootstrap/bootstrap.py:920]: https://github.com/rust-lang/rust/pull/113637

Comment (ozkanonur) [reply src/bootstrap/bootstrap.py:920]: just to be sure; does that `FIXME` means applying the flag unconditionally or not setting it at all anymore ?

cc @jyn514

**Comment by ozkanonur** [comment]
@bors r-

I want to be sure on this https://github.com/rust-lang/rust/pull/113683#discussion_r1263900648

File: src/bootstrap/suggest.rs [hunk 3]
```diff
@@ -67,10 +67,14 @@ pub fn suggest(builder: &Builder<'_>, run: bool) {
 
     if run {
         for sug in suggestions {
+            let (cmd, stage, paths) = sug;
```
Comment (jyn514) [inline src/bootstrap/suggest.rs:70]:
```suggestion
        for (cmd, stage, paths) in suggestions {
```

File: src/bootstrap/bootstrap.py (hunk 2 above)
Comment (jyn514) [reply src/bootstrap/bootstrap.py:920]: this is part of the target-spec now, so you can remove it altogether: https://github.com/rust-lang/rust/blob/bacf5bcbc7dae526872cf6af3fc0064a51b1efa2/compiler/rustc_target/src/spec/mips64_unknown_linux_gnuabi64.rs#L15
```suggestion
```

File: src/bootstrap/builder.rs [hunk 4]
```diff
@@ -903,17 +903,7 @@ impl<'a> Builder<'a> {
...
-        paths: Vec<PathBuf>,
-        stage: Option<u32>,
-    ) -> Builder<'_> {
-        // FIXME: don't mutate `build`
-        if let Some(stage) = stage {
-            build.config.stage = stage;
-        }
-
+    pub fn new_standalone(build: &Build, kind: Kind, paths: Vec<PathBuf>) -> Builder<'_> {
         Self::new_internal(build, kind, paths.to_owned())
```
Comment (jyn514) [inline src/bootstrap/builder.rs:907]: at point there's no reason to have a separate `standalone` function and you can just make `new_internal` public.

this function is going away altogether in https://github.com/rust-lang/rust/pull/113644/commits/79c670037617df68097d68d9f81708b1c96bab75 though, so maybe just drop this commit for now?

File: src/bootstrap/suggest.rs (hunk 3 above)
Comment (ozkanonur) [reply src/bootstrap/suggest.rs]: Related commit is dropped from PR

**Comment by ozkanonur** [comment]
Hmm. Failed on trying to download broken link https://ci-artifacts.rust-lang.org/rustc-builds-alt/bef6ff618f17398775e9d8cd23a6f47d983869dc/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz 

@rustbot retry

**Comment by ozkanonur** [comment]
Hmm. Failed on trying to download broken link https://ci-artifacts.rust-lang.org/rustc-builds-alt/bef6ff618f17398775e9d8cd23a6f47d983869dc/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz 

@bors retry

**Pull request merged**